polaris-patterns/src/components/PolarisPatternsProvider/PolarisPatternsProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import type {PropsWithChildren} from 'react';
+import React, {createContext, useEffect} from 'react';
+import {I18nextProvider, initReactI18next} from 'react-i18next';
+import {createInstance} from 'i18next';
+
+import {DEFAULT_I18N_DETAILS} from '../../configure';
+
+export const I18nDetails = createContext(DEFAULT_I18N_DETAILS);
+
+const i18next = createInstance({
+  fallbackLng: DEFAULT_I18N_DETAILS.locale,
+  interpolation: {
+    escapeValue: false,
+  },
+  resources: {},
+  returnNull: false,
+});
+
+i18next.use(initReactI18next).init();
+
+interface TranslationDictionary {
+  [key: string]: string | TranslationDictionary;
+}
+
+interface I18nDetails {
+  locale: string;
+  country?: string;
+  currency?: string;
+  timezone?: string;
+}
+
+export interface Props {
+  i18nDetails: I18nDetails;
+  translations: TranslationDictionary;
+  enFallback: TranslationDictionary;
+}
+
+export function PolarisPatternsProvider({
+  translations,
+  i18nDetails,
+  enFallback,
+  children,
+}: PropsWithChildren<Props>) {
+  useEffect(() => {
+    i18next.addResourceBundle('en', 'translation', enFallback);
+  }, [enFallback]);
+  useEffect(() => {
+    i18next.changeLanguage(i18nDetails.locale);
+    i18next.addResourceBundle(i18nDetails.locale, 'translation', translations);
```

Comment:
You should call `changeLanguage` after `addResourceBundle` so that the translations are present when the language is changed.

```suggestion
    i18next.addResourceBundle(i18nDetails.locale, 'translation', translations);
    i18next.changeLanguage(i18nDetails.locale);
```